rero_ils/alembic/bd78d77eb7e3_isfiction.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2022 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Add fiction."""
+
+from logging import getLogger
+
+from rero_ils.modules.documents.api import Document, DocumentsSearch
+
+# revision identifiers, used by Alembic.
+revision = 'bd78d77eb7e3'
+down_revision = ('fc45b1b998b8', 'a941628259e1')
+branch_labels = ()
+depends_on = None
+
+LOGGER = getLogger('alembic')
+
+
+def upgrade():
+    """Upgrade database."""
+    FICTIONS_TERMS = ['Fictions', 'Films de fiction']
+    query = DocumentsSearch() \
+        .filter('terms', facet_genre_form_en=FICTIONS_TERMS)
+    LOGGER.info(f'Add fiction=true to documents: {query.count()}')
+    for idx, hit in enumerate(query.source('pid').scan(), 1):
+        if (doc := Document.get_record_by_pid(hit.pid)):
```

Comment:
You can rewrite like this
```python
query = DocumentsSearch() \
        .filter('terms', facet_genre_form_en=FICTIONS_TERMS)
LOGGER.info(f'Add fiction=true to documents: {query.count()}')
ids = [hit.meta.id in hit for query.source().scan()]
LOGGER.info(f'Add fiction=true to documents: {query.count()}')
for idx, _id in enumerate(ids, 1):
  if (doc := Document.get_record(_id)):
```
in this way you prevent elasticsearch timeout and you save many database calls.
for reminder: with `get_record_by_pid` does 2 calls to retreive record and `get_record` does one call.